reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/job/parameter/JobParameter.java

```diff
@@ -0,0 +1,148 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.job.parameter;
+
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.ScheduledJobParameter;
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.Schedule;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+
+import java.io.IOException;
+import java.time.Instant;
+
+/**
+ * A sample job parameter.
+ * <p>
+ * It adds an additional "indexToWatch" field to {@link ScheduledJobParameter}, which stores the index
+ * the job runner will watch.
+ */
+public class JobParameter implements ScheduledJobParameter {
```

Comment:
why is this mutable? is it possible to get rid of setters?

Reply:
made variables final and removed setters